emily/handler/src/context.rs

```diff
@@ -73,4 +73,66 @@ impl EmilyContext {
             dynamodb_client: Client::new(&config),
         })
     }
+    /// Create a local testing instance.
+    #[cfg(feature = "testing")]
+    pub async fn local_test_instance() -> Result<Self, Error> {
+        use std::collections::HashMap;
+
+        // Get config that always points to the dynamodb table directly
+        // from outside of a docker compose setup.
+        let dynamodb_client = Client::new(
+            &aws_config::load_defaults(BehaviorVersion::latest())
+                .await
+                .into_builder()
+                .endpoint_url("http://localhost:8000")
+                .build(),
+        );
```

Comment:
nit: can we break this out like so? What's here isn't so bad, but I find breakout approach a bit easier to read
```rust
let sdk_config = aws_config::load_defaults(BehaviorVersion::latest())
    .await
    .into_builder()
    .endpoint_url("http://localhost:8000")
    .build();
let dynamodb_client = Client::new(&sdk_config);
```

Reply:
Do you have any specific "rule" for when to break variables out? Could help me calibrate. My personal rule has typically been "if it can be done without semicolons, it should be done without semicolons" but I agree that it's leading to unclear code so I should change that.

Reply:
Yeah, my rough "rule" is something like "assign 'computation outputs' to a new variable unless 'the whole thing' can fit in one line". So I might do something like this
```rust
// This is simple enough
let dynamodb_client = Client::new(&make_my_config());
```
But it's a rough rule, since I probably wouldn't do the following even if it fits on one line:
```rust
// There is too much going on here. Also, clippy might change
// this anyway, and if it does then I'd break it out
let dynamodb_client = Client::new(&ConfigBuilder::new().endpoint(url).build());
```
I don't expect anyone to follow my "rule" but when I think that other people will find it much easier to read then I'll nit on it.